src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -286,45 +291,769 @@ public List<Element> getMembers(Kind kind) {
+            assert !m.containsKey(e1);
+            m.put(e1, m.size());
+        }
+        // patch for interfaces
+        if (e.getKind().isInterface()) {
+            // java.lang.Object is the last supertype in a graph of supertypes
+            // for an interface
+            var prev = m.put(elements().getTypeElement("java.lang.Object"), m.size());
+            assert prev == null;
+        }
+        return m;
+    }
+
+    public Set<TypeMirror> getAllInterfaces(TypeElement te) {
+        Set<TypeMirror> results = new LinkedHashSet<>();
+        addSuperInterfaces(te.asType(), results, new HashSet<>());
+        return results;
+    }
+
+    private void addSuperInterfaces(TypeMirror type, Set<TypeMirror> results, Set<Element> visited) {
+        TypeMirror superType = null;
+        for (TypeMirror t : types().directSupertypes(type)) {
+            if (types().isSameType(t, utils.getObjectType()))
+                continue;
+            TypeElement e = asTypeElement(t);
+            if (utils.isPlainInterface(e)) {
+                if (!visited.add(e)) {
+                    continue; // seen it before
+                }
+                results.add(t);
+                addSuperInterfaces(t, results, visited);
+            } else {
+                // there can be at most one superclass and it is not null
+                assert superType == null && t != null : superType;
+                // Save the supertype for later.
+                superType = t;
+            }
+        }
+        // Collect the super-interfaces of the supertype.
+        if (superType != null)
+            addSuperInterfaces(superType, results, visited);
+    }
+
+
+
+    private com.sun.tools.javac.util.List<OverrideData> findOverriddenBy(ExecutableElement m) {
+        return findOverriddenBy(m, (DeclaredType) te.asType(), com.sun.tools.javac.util.List.nil(), new HashSet<>());
+    }
+
+    private com.sun.tools.javac.util.List<OverrideData> findOverriddenBy(ExecutableElement m,
+                                                                         DeclaredType declaredType,
+                                                                         com.sun.tools.javac.util.List<OverrideData> result,
+                                                                         Set<ExecutableElement> foundOverriddenSoFar) {
+        for (var s : directSupertypes(declaredType)) {
+            var supertypeElement = (TypeElement) s.asElement();
+            for (var m1 : mcache.getVisibleMemberTable(supertypeElement).getMethodMembers())
+                if (utils.elementUtils.overrides(m, m1, te) && foundOverriddenSoFar.add(m1)) {
+                    // use any value for `simpleOverride` for now: the actual value
+                    // will be computed once the complete sequence is available
+                    var simpleOverride = false;
+                    result = result.append(new OverrideData(toDeclaringType(s,
```

Comment:
`append` is *horrible* on `javac.List`

Unless there is a string reason to be using it, you should probably be using plain old collection Lists, like `ArrayList` or `List.of`. 

If you're using `javac.List` because you're calling javac internal functionality that requires that kind of List, then your usage should be in methods in `WorkArounds`, which is intended to the the only doclet class that accesses javac internal methods and types.

Reply:
Replied here https://github.com/openjdk/jdk/pull/14221#discussion_r1224059741